compiler/rustc_lint/src/traits.rs

```diff
@@ -141,3 +141,30 @@ impl<'tcx> LateLintPass<'tcx> for DropTraitConstraints {
         }
     }
 }
+
+declare_lint! {
+    pub DUP_TRAIT_BOUNDS,
+    Deny,
+    "duplicate trait bounds"
+}
+
+declare_lint_pass!(
+    DuplicateTraitBounds => [DUP_TRAIT_BOUNDS]
+);
+
+impl<'tcx> LateLintPass<'tcx> for DuplicateTraitBounds {
+    fn check_ty(&mut self, cx: &LateContext<'_>, ty: &'tcx hir::Ty<'tcx>) {
+        let hir::TyKind::TraitObject(bounds, _lifetime, _syntax) = &ty.kind else {
+            return
+        };
+
+        let bounds: Vec<_> = bounds.into_iter().collect();
+
+        for bound in bounds {
+            cx.struct_span_lint(DUP_TRAIT_BOUNDS, bound.span, |lint| {
+                let msg = format!("There's a duplicate trait bound here");
+                lint.build(&msg).emit()
+            });
```

Comment:
Just trying to get the basics of how this code should work down.

With the sample program [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=62c189f3c882cffe1e91e7511d4a0179), I expected invoking `rustc` to emit a bunch of errors. But it reported nothing...